### PR TITLE
refactor(go)!: hide internals of ffi communication and provide cleaner client

### PR DESCRIPTION
**Pull request by erka**
This is a breaking change on API level. 
As it's still v0 I think it's okay to make this change

### CODE REVIEW

**Comment by erka** [comment]
@markphelps this is PR into your pull request

File: flipt-client-go/evaluation.go [hunk 1]
```diff
@@ -157,11 +163,15 @@ func (e *Client) EvaluateBoolean(_ context.Context, flagKey, entityID string, ev
 		return nil, err
 	}
 
-	return br, nil
+	if br.Status == statusSuccess {
```
Comment (markphelps) [inline flipt-client-go/evaluation.go:166]: we could make this a non breaking API change by returning `br` here and still check if `br.Status == success` and return an error if not right?

i just worry about making this breaking change in only one sdk until we have tried it in all client side sdks to make sure it is the 'shape' of the responses that we want.

wdyt?

**Review by markphelps — COMMENTED** [review]
thank you! I do have a question on if we should make this breaking change now or do another pass of all the client side sdk's to make sure we are somewhat consistent if we are going to change the response types

File: flipt-client-go/evaluation.go (hunk 1 above)
Comment (erka) [reply flipt-client-go/evaluation.go:166]: yes, we can do that.  Please revert those or I will do it tomorrow.

**Pull request merged**